File: xchange-idexjaxrs/src/main/java/org/knowm/xchange/idex/IdexWithdraw.java
package org.knowm.xchange.idex;

import org.jetbrains.annotations.NotNull;
import org.knowm.xchange.idexjaxrs.dto.WithdrawReq;
import org.knowm.xchange.service.trade.params.WithdrawFundsParams;

import java.math.BigInteger;

final class IdexWithdraw extends WithdrawReq implements WithdrawFundsParams {
    public IdexWithdraw(@NotNull String address, @NotNull String amount, @NotNull String token, @NotNull BigInteger nonce, @NotNull String s, @NotNull BigInteger v, @NotNull String r) {
       this.address(address).amount(amount).token(token).nonce(nonce).s(s).v(v).r(r);
    }
}

File: xchange-idexjaxrs/src/main/kotlin/IdexWithdraw.kt
package org.knowm.xchange.idexjaxrs


import org.knowm.xchange.idexjaxrs.dto.*
import org.knowm.xchange.service.trade.params.*
import java.math.*

class IdexWithdraw(address: String, amount: String, token: String, nonce: BigInteger, s: String, v: BigInteger,
                   r: String) : WithdrawReq(), WithdrawFundsParams {init {
    address(address)
            .amount(amount)
            .token(token)
            .nonce(nonce)
            .s(s)
            .v(v)
            .r(r)
}
}

